### ISSUE TITLE
Fix S2953 FP: Implement `IDisposable.Dispose` on extension methods `static` class

### ISSUE THREAD

**Issue by manfred-brands** [SonarSource/sonar-dotnet#9679]
### Description

S2953 raised a false positive on a `static` class.

### Repro steps

We have an extension method on `Lazy` to dispose it, but only if the value was created.

```csharp
    public static class DisposableExtensions
    {
        public static void Dispose<T>(this Lazy<T> lazy)
            where T : class, IDisposable
        {
            if (lazy.IsValueCreated)
            {
                lazy.Value.Dispose();
            }
        }
    }
```

### Expected behavior

S2953 is not raised.

### Actual behavior

S2953 is raised.

### Known workarounds

Suppress

### Related information

* C#/VB.NET Plugins version:  9.32.0.97167
* Visual Studio version: 17.11.4
* MSBuild / dotnet version: 8.0.402
* SonarScanner for .NET version (if used)
* Operating System: Win10 22H2

**Comment by CristianAmbrosini** [comment]
Hi @manfred-brands!
I confirm this as a FP.
I already added the reproducer and fixed the issue; it was a low-hanging fruit.
You will not experience it anymore starting from the next release, enjoy!

**Comment by manfred-brands** [comment]
@CristianAmbrosini Thanks, but I don't see neither a PR or changes to master with a fix so how can this be closed?
Did you forgot to push your changes?
